unlock-app/src/services/walletService.js

```diff
@@ -91,6 +94,35 @@ export default class WalletService extends EventEmitter {
     }
   }
 
+  /**
+   * Poll to see if account has changed
+   */
+  async pollForAccountChange(timeout, isServer) {
+    if (isServer) return
+    while (true) {
```

Comment:
Not a huge fan of `while (true)`.

Couldn't we just call `pollForAccountChange` after we have waited for the stuff we care about?

Reply:
Perhaps a little clever `await this.pollAccountChange()` at the end of the function? Although this will kill the call stack eventually. Hmm... Let think on this